app/services/registration_checker.rb

```diff
@@ -200,5 +216,26 @@ def existing_registration_in_series?
       end
       false
     end
+
+    def competitor_qualifies_for_event?(event, qualification)
+      competitor_qualification_results = UserApi.qualifications(@requestee_user_id)
+      result_type = qualification['resultType']
+
+      competitor_pr = competitor_qualification_results.find { |result| result['eventId'] == event && result['type'] == result_type }
+      return false if competitor_pr.blank?
+
+      return false unless Date.parse(competitor_pr['on_or_before']) <= Date.parse(qualification['whenDate'])
+
+      case qualification['type']
+      when 'anyResult', 'ranking'
+        # By this point the user definitely has a result.
+        # Ranking qualifications are enforced when registration closes, so it is effectively an anyResult ranking when registering
+        true
+      when 'attemptResult'
+        competitor_pr['best'].to_i < qualification['level']
```

Comment:
Attention! This is not trivial for FMC and MultiBLD. I think you need to properly parse the result instead of just using `to_i` and calling it a day.

Reply:
As discussed, straight integer comparison is fine